examples/sample_trading_strategy.py

```diff
@@ -0,0 +1,133 @@
+                instrument = alias_to_instrument[alias]
+                instrument_pips = instrument["pips"]
+                print("pips = " + str(instrument_pips), flush=True)
+                # note: alias_to_position[alias] is not actual position size, actual position size = alias_to_position[alias] * size_multiplier
+                position = 0
+                if (alias in alias_to_position):
+                    position = alias_to_position[alias]
+                # note: position is not actual position size, actual position size = position * size_multiplier
+                if (position < INITIAL_THRESHOLD):
+                    if (alias in alias_to_buy_order_id):
+                        print("attempt to move buy order with price" + str(best_bid_price_level * instrument_pips - INITIAL_SHIFT), flush=True)
+                        bm.move_order(addon, alias, alias_to_buy_order_id[alias], best_bid_price_level * instrument_pips - INITIAL_SHIFT)
+                    else:
+                        print("attempt to send buy order with price" + str(best_bid_price_level * instrument_pips - INITIAL_SHIFT), flush=True)
+                        # note: INITIAL_ORDER_SIZE is not actual order size, actual order size = INITIAL_ORDER_SIZE * size_multiplier
+                        bm.send_order(addon, alias, True, INITIAL_ORDER_SIZE, 'GTC', best_bid_price_level * instrument_pips - INITIAL_SHIFT)
+
+                # note: position is not actual position size, actual position size = position * size_multiplier
+                if (position > -INITIAL_THRESHOLD):
+                    if (alias in alias_to_sell_order_id):
+                        print("attempt to move sell order with price" + str(best_ask_price_level * instrument_pips + INITIAL_SHIFT), flush=True)
+                        bm.move_order(addon, alias, alias_to_sell_order_id[alias], best_ask_price_level * instrument_pips + INITIAL_SHIFT)
+                    else:
+                        print("attempt to send sell order with price" + str(best_ask_price_level * instrument_pips + INITIAL_SHIFT), flush=True)
+                        # note: INITIAL_ORDER_SIZE is not actual order size, actual order size = INITIAL_ORDER_SIZE * size_multiplier
+                        bm.send_order(addon, alias, False, INITIAL_ORDER_SIZE, 'GTC', best_ask_price_level * instrument_pips + INITIAL_SHIFT)
+
+
+if __name__ == "__main__":
+    addon = bm.create_addon()
+    bm.add_depth_handler(addon, handle_depth_info)
+    bm.add_on_interval_handler(addon, on_interval)
+    bm.add_on_order_executed_handler(addon, order_execute_handler)
+    bm.add_on_order_updated_handler(addon, order_update_handler)
+    bm.add_on_position_update_handler(addon, position_update_handler)
+    bm.start_addon(addon, handle_instrument_info, handle_instrument_detached)
+    bm.wait_until_addon_is_turned_off(addon)
```

Comment:
Do we have some dialog for trading strategies on the java side which explains that usage of this code may cause troubles if you trade on real account. I believe it should ask user to agree that he understands risks. 

So it should be similar to how other trading addons are done.

Reply:
Yes, we have such dialog. If an addon has annotation @Layer1TradingStrategy then the user should approve that he/she understands risks and confirms all further trading activity with this addon. 
It is a part of core api.